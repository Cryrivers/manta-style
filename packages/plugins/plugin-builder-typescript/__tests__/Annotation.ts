import getTranspiledString from '../src/utils/transpile';
import { BuilderPluginTestHelper } from '@manta-style/test-helpers';

const testTranspiledString = BuilderPluginTestHelper.testTranspiledString(
  getTranspiledString,
);

describe('Annotation', () => {
  test('Annotation', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin a b c }}
       */
      a: string;
      /**
       * @mantastyle {{ multiline a
       *  b
       *  c }}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });
  test('Handles { and }, escaped bracket', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin "{{foo}}" }}
       */
      a: string;
      /**
       * @mantastyle {{ plugin "{{foo}} {{bar}}" }}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });

  test('Handles }} in other annotation', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin "{{foo}}" }}
       * @random {{}}
       */
      a: string;
      /**
       * @mantastyle {{ plugin "{{foo}} {{bar}}" }}
       * @random {{}}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });

  test('Handles random }} in next row', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin "{{foo}}" }}
       * }}
       */
      a: string;
      /**
       * @mantastyle {{ multiline 
       *   "{{foo}} {{bar}}"
       *   123
       *   (as "ew")
       * }}
       * }}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });

  test('Handles hash', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin "{{foo}}" bar="baz" }}
       */
      a: string;
      /**
       * @mantastyle {{ multiline 
       *   "{{foo}} {{bar}}"
       *   123
       *   (as "ew" foo=(bar "foo" baz="bar"))
       * }}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });
});
