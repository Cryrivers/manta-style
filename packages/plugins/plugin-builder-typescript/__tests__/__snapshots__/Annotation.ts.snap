// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotation Annotation 1`] = `
"FROM:
type A = {
      /**
       * @mantastyle {{ plugin a b c }}
       */
      a: string;
      /**
       * @mantastyle {{ multiline a
       *  b
       *  c }}
       */
      b: number;
    };

--------------------------------------------------------
TO:
'use strict';
exports.__esModule = true;
var runtime_1 = require('@manta-style/runtime');
var A = runtime_1['default'].TypeAliasDeclaration(
  'A',
  function(typeFactory) {
    var type = runtime_1['default'].TypeLiteral(function(typeLiteral) {
      typeLiteral.property(
        'a',
        runtime_1['default'].StringKeyword,
        false,
        runtime_1['default'].MantaAnnotation('{{ plugin a b c }}'),
      );
      typeLiteral.property(
        'b',
        runtime_1['default'].NumberKeyword,
        false,
        runtime_1['default'].MantaAnnotation('{{ multiline a\\\\nb\\\\nc }}'),
      );
    });
    return type;
  },
  runtime_1['default'].MantaAnnotation(''),
);
"
`;

exports[`Annotation Handles { and }, escaped bracket 1`] = `
"FROM:
type A = {
      /**
       * @mantastyle {{ plugin \\"{{foo}}\\" }}
       */
      a: string;
      /**
       * @mantastyle {{ plugin \\"{{foo}} {{bar}}\\" }}
       */
      b: number;
    };

--------------------------------------------------------
TO:
'use strict';
exports.__esModule = true;
var runtime_1 = require('@manta-style/runtime');
var A = runtime_1['default'].TypeAliasDeclaration(
  'A',
  function(typeFactory) {
    var type = runtime_1['default'].TypeLiteral(function(typeLiteral) {
      typeLiteral.property(
        'a',
        runtime_1['default'].StringKeyword,
        false,
        runtime_1['default'].MantaAnnotation('{{ plugin \\"{{foo}}\\" }}'),
      );
      typeLiteral.property(
        'b',
        runtime_1['default'].NumberKeyword,
        false,
        runtime_1['default'].MantaAnnotation('{{ plugin \\"{{foo}} {{bar}}\\" }}'),
      );
    });
    return type;
  },
  runtime_1['default'].MantaAnnotation(''),
);
"
`;

exports[`Annotation Handles }} in other annotation 1`] = `
"FROM:
type A = {
      /**
       * @mantastyle {{ plugin \\"{{foo}}\\" }}
       * @random {{}}
       */
      a: string;
      /**
       * @mantastyle {{ plugin \\"{{foo}} {{bar}}\\" }}
       * @random {{}}
       */
      b: number;
    };

--------------------------------------------------------
TO:
'use strict';
exports.__esModule = true;
var runtime_1 = require('@manta-style/runtime');
var A = runtime_1['default'].TypeAliasDeclaration(
  'A',
  function(typeFactory) {
    var type = runtime_1['default'].TypeLiteral(function(typeLiteral) {
      typeLiteral.property(
        'a',
        runtime_1['default'].StringKeyword,
        false,
        runtime_1['default'].MantaAnnotation('{{ plugin \\"{{foo}}\\" }}'),
      );
      typeLiteral.property(
        'b',
        runtime_1['default'].NumberKeyword,
        false,
        runtime_1['default'].MantaAnnotation('{{ plugin \\"{{foo}} {{bar}}\\" }}'),
      );
    });
    return type;
  },
  runtime_1['default'].MantaAnnotation(''),
);
"
`;

exports[`Annotation Handles random }} in next row 1`] = `
"FROM:
type A = {
      /**
       * @mantastyle {{ plugin \\"{{foo}}\\" }}
       * }}
       */
      a: string;
      /**
       * @mantastyle {{ multiline 
       *   \\"{{foo}} {{bar}}\\"
       *   123
       *   (as \\"ew\\")
       * }}
       * }}
       */
      b: number;
    };

--------------------------------------------------------
TO:
'use strict';
exports.__esModule = true;
var runtime_1 = require('@manta-style/runtime');
var A = runtime_1['default'].TypeAliasDeclaration(
  'A',
  function(typeFactory) {
    var type = runtime_1['default'].TypeLiteral(function(typeLiteral) {
      typeLiteral.property(
        'a',
        runtime_1['default'].StringKeyword,
        false,
        runtime_1['default'].MantaAnnotation('{{ plugin \\"{{foo}}\\" }}'),
      );
      typeLiteral.property(
        'b',
        runtime_1['default'].NumberKeyword,
        false,
        runtime_1['default'].MantaAnnotation(
          '{{ multiline \\\\n\\"{{foo}} {{bar}}\\"\\\\n123\\\\n(as \\"ew\\")\\\\n}}',
        ),
      );
    });
    return type;
  },
  runtime_1['default'].MantaAnnotation(''),
);
"
`;
